### ISSUE TITLE
Build broken on Ubuntu Xenial: Octave parse error in duration near "days = 0 % planar"

### ISSUE THREAD

**Issue by apjanke** [apjanke/octave-chrono#26]
https://travis-ci.com/apjanke/octave-chrono/builds/98352408

```
$ make test
octave --no-gui --silent --norc --eval="octave_chrono_make_local"
Built __oct_time_binsearch__
./dev-tools/runtests.sh inst
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
Processing files in /home/travis/build/apjanke/octave-chrono/inst:
  datetime.m ............................................. PASS    0/4    FAIL 4
  duration.m ............................................. PASS    0/3    FAIL 3
The following files in /home/travis/build/apjanke/octave-chrono/inst have no tests:
NaT.m               days.m              milliseconds.m      years.m
calendarDuration.m  hours.m             minutes.m
calmonths.m         isdatetime.m        seconds.m
calyears.m          isduration.m        timezones.m
runtests.sh: Some tests FAILED!
runtests.sh: running additional tests so we can get output
runtests.sh: testing datetime
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
***** test datetime;
!!!!! test failed
parse error near line 82 of file /home/travis/build/apjanke/octave-chrono/inst/datetime.m
  syntax error
>>>     dnums = NaN % planar
                            ^
runtests.sh: testing duration
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
***** test duration;
!!!!! test failed
parse error near line 59 of file /home/travis/build/apjanke/octave-chrono/inst/duration.m
  syntax error
>>>     days = 0 % planar
```

That's on Octave 4.0.0. Might need to install a newer Octave in our CI, like we're [doing with octave-jsonstuff](https://github.com/apjanke/octave-jsonstuff/blob/f2a08eb3afec3dce5fb19e73ec0e263b1d2f38ba/dev-tools/install-octave-4.2-ubuntu.sh).

**Comment by apjanke** [comment]
Upgraded to Octave 4.2. Still a build failure.

```
$ make test
octave --no-gui --silent --norc --eval="octave_chrono_make_local"
Built __oct_time_binsearch__
./dev-tools/runtests.sh inst
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
Processing files in /home/travis/build/apjanke/octave-chrono/inst:
  datetime.m ............................................. PASS    0/4    FAIL 4
  duration.m ............................................. PASS    0/3    FAIL 3
The following files in /home/travis/build/apjanke/octave-chrono/inst have no tests:
NaT.m               days.m              milliseconds.m      years.m
calendarDuration.m  hours.m             minutes.m
calmonths.m         isdatetime.m        seconds.m
calyears.m          isduration.m        timezones.m
runtests.sh: Some tests FAILED!
runtests.sh: running additional tests so we can get output
runtests.sh: testing datetime
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
***** test datetime;
!!!!! test failed
parse error near line 999 of file /home/travis/build/apjanke/octave-chrono/inst/datetime.m
  syntax error
>>> function out = promotec (args)
           ^
runtests.sh: testing duration
octave: X11 DISPLAY environment variable not set
octave: disabling GUI features
***** test duration;
!!!!! test failed
parse error near line 789 of file /home/travis/build/apjanke/octave-chrono/inst/duration.m
  syntax error
>>> function varargout = promote (varargin)
           ^
Makefile:174: recipe for target 'test' failed
make: *** [test] Error 1
The command "make test" exited with 2.
```

Test this out locally on Xenial before doing much more on CI.

**Comment by apjanke** [comment]
Looks like local functions in classdef files were only introduced in Octave 4.4.0.

https://savannah.gnu.org/bugs/?41723

https://hg.savannah.gnu.org/hgweb/octave/rev/67a638ac7f24

That changeset is from April 3 2017. 4.4.0 was tagged on April 30 2018.

**Comment by apjanke** [comment]
Fixed by refactoring for Octave 4.2 compatibility in https://github.com/apjanke/octave-chrono/commit/29b89a66ed3b39aecaaadf5322d26f2dbad70c7c.

CI is green now: https://travis-ci.com/apjanke/octave-chrono/builds/98371302.
